// @algorandfoundation/algorand-typescript/gtxn-CXBH4yay.js
import { N as NoImplementation } from './errors-D124-zqo.js';

/**
 * Get the nth transaction in the group without verifying its type
 * @param n The index of the txn in the group
 */
function Transaction(n) {
    throw new NoImplementation();
}
/**
 * Get the nth transaction in the group
 * Verifies the txn type is 'pay'
 * @param n The index of the txn in the group
 */
function PaymentTxn(n) {
    throw new NoImplementation();
}
/**
 * Get the nth transaction in the group
 * Verifies the txn type is 'keyreg'
 * @param n The index of the txn in the group
 */
function KeyRegistrationTxn(n) {
    throw new NoImplementation();
}
/**
 * Get the nth transaction in the group
 * Verifies the txn type is 'acfg'
 * @param n The index of the txn in the group
 */
function AssetConfigTxn(n) {
    throw new NoImplementation();
}
/**
 * Get the nth transaction in the group
 * Verifies the txn type is 'axfer'
 * @param n The index of the txn in the group
 */
function AssetTransferTxn(n) {
    throw new NoImplementation();
}
/**
 * Get the nth transaction in the group
 * Verifies the txn type is 'afrz'
 * @param n The index of the txn in the group
 */
function AssetFreezeTxn(n) {
    throw new NoImplementation();
}
/**
 * Get the nth transaction in the group
 * Verifies the txn type is 'appl'
 * @param n The index of the txn in the group
 */
function ApplicationCallTxn(n) {
    throw new NoImplementation();
}

var gtxn = /*#__PURE__*/Object.freeze({
    __proto__: null,
    ApplicationCallTxn: ApplicationCallTxn,
    AssetConfigTxn: AssetConfigTxn,
    AssetFreezeTxn: AssetFreezeTxn,
    AssetTransferTxn: AssetTransferTxn,
    KeyRegistrationTxn: KeyRegistrationTxn,
    PaymentTxn: PaymentTxn,
    Transaction: Transaction
});

export { AssetConfigTxn as A, KeyRegistrationTxn as K, PaymentTxn as P, Transaction as T, AssetTransferTxn as a, AssetFreezeTxn as b, ApplicationCallTxn as c, gtxn as g };
