// @algorandfoundation/algorand-typescript/itxn-CPeSA_Q_.js
import { N as NoImplementation } from './errors-D124-zqo.js';

/**
 * Submit a group of ItxnParams objects and return the InnerTransaction results
 */
function submitGroup(...transactionFields) {
    throw new NoImplementation();
}
/**
 * Holds Payment fields which can be updated, cloned, or submitted.
 */
class PaymentItxnParams {
    /**
     * Submit an itxn with these fields and return the PaymentInnerTxn result
     */
    submit() {
        throw new NoImplementation();
    }
    /**
     * Update one or more fields in this PaymentItxnParams object
     */
    set(fields) {
        throw new NoImplementation();
    }
    /**
     * Return a copy of this PaymentItxnParams object
     */
    copy() {
        throw new NoImplementation();
    }
}
/**
 * Create a new PaymentItxnParams object with the specified fields
 */
function payment(fields) {
    throw new NoImplementation();
}
/**
 * Holds KeyRegistration fields which can be updated, cloned, or submitted.
 */
class KeyRegistrationItxnParams {
    /**
     * Submit an itxn with these fields and return the KeyRegistrationInnerTxn result
     */
    submit() {
        throw new NoImplementation();
    }
    /**
     * Update one or more fields in this KeyRegistrationItxnParams object
     */
    set(fields) {
        throw new NoImplementation();
    }
    /**
     * Return a copy of this KeyRegistrationItxnParams object
     */
    copy() {
        throw new NoImplementation();
    }
}
/**
 * Create a new KeyRegistrationItxnParams object with the specified fields
 */
function keyRegistration(fields) {
    throw new NoImplementation();
}
/**
 * Holds AssetConfig fields which can be updated, cloned, or submitted.
 */
class AssetConfigItxnParams {
    /**
     * Submit an itxn with these fields and return the AssetConfigInnerTxn result
     */
    submit() {
        throw new NoImplementation();
    }
    /**
     * Update one or more fields in this AssetConfigItxnParams object
     */
    set(fields) {
        throw new NoImplementation();
    }
    /**
     * Return a copy of this AssetConfigItxnParams object
     */
    copy() {
        throw new NoImplementation();
    }
}
/**
 * Create a new AssetConfigItxnParams object with the specified fields
 */
function assetConfig(fields) {
    throw new NoImplementation();
}
/**
 * Holds AssetTransfer fields which can be updated, cloned, or submitted.
 */
class AssetTransferItxnParams {
    /**
     * Submit an itxn with these fields and return the AssetTransferInnerTxn result
     */
    submit() {
        throw new NoImplementation();
    }
    /**
     * Update one or more fields in this AssetTransferItxnParams object
     */
    set(fields) {
        throw new NoImplementation();
    }
    /**
     * Return a copy of this AssetTransferItxnParams object
     */
    copy() {
        throw new NoImplementation();
    }
}
/**
 * Create a new AssetTransferItxnParams object with the specified fields
 */
function assetTransfer(fields) {
    throw new NoImplementation();
}
/**
 * Holds AssetFreeze fields which can be updated, cloned, or submitted.
 */
class AssetFreezeItxnParams {
    /**
     * Submit an itxn with these fields and return the AssetFreezeInnerTxn result
     */
    submit() {
        throw new NoImplementation();
    }
    /**
     * Update one or more fields in this AssetFreezeItxnParams object
     */
    set(fields) {
        throw new NoImplementation();
    }
    /**
     * Return a copy of this AssetFreezeItxnParams object
     */
    copy() {
        throw new NoImplementation();
    }
}
/**
 * Create a new AssetFreezeItxnParams object with the specified fields
 */
function assetFreeze(fields) {
    throw new NoImplementation();
}
/**
 * Holds ApplicationCall fields which can be updated, cloned, or submitted.
 */
class ApplicationCallItxnParams {
    /**
     * Submit an itxn with these fields and return the ApplicationCallInnerTxn result
     */
    submit() {
        throw new NoImplementation();
    }
    /**
     * Update one or more fields in this ApplicationCallItxnParams object
     */
    set(fields) {
        throw new NoImplementation();
    }
    /**
     * Return a copy of this ApplicationCallItxnParams object
     */
    copy() {
        throw new NoImplementation();
    }
}
/**
 * Create a new ApplicationCallItxnParams object with the specified fields
 */
function applicationCall(fields) {
    throw new NoImplementation();
}

var itxn = /*#__PURE__*/Object.freeze({
    __proto__: null,
    ApplicationCallItxnParams: ApplicationCallItxnParams,
    AssetConfigItxnParams: AssetConfigItxnParams,
    AssetFreezeItxnParams: AssetFreezeItxnParams,
    AssetTransferItxnParams: AssetTransferItxnParams,
    KeyRegistrationItxnParams: KeyRegistrationItxnParams,
    PaymentItxnParams: PaymentItxnParams,
    applicationCall: applicationCall,
    assetConfig: assetConfig,
    assetFreeze: assetFreeze,
    assetTransfer: assetTransfer,
    keyRegistration: keyRegistration,
    payment: payment,
    submitGroup: submitGroup
});

export { AssetConfigItxnParams as A, KeyRegistrationItxnParams as K, PaymentItxnParams as P, assetConfig as a, AssetTransferItxnParams as b, assetTransfer as c, AssetFreezeItxnParams as d, assetFreeze as e, ApplicationCallItxnParams as f, applicationCall as g, itxn as i, keyRegistration as k, payment as p, submitGroup as s };
